src/bootstrap/test.rs

```diff
@@ -827,6 +827,46 @@ impl Step for RustdocJSNotStd {
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct RustdocJSParser {
+    pub target: TargetSelection,
+}
+
+impl Step for RustdocJSParser {
+    type Output = ();
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/test/rustdoc-js-parser")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(RustdocJSParser { target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) {
+        if let Some(ref nodejs) = builder.config.nodejs {
+            let mut command = Command::new(nodejs);
+            command
+                .arg(builder.src.join("src/tools/rustdoc-js/tester.js"))
+                .arg("--test-parser") // needed to run parser checks instead of search tests
+                .arg("--crate-name")
+                .arg("std")
+                .arg("--resource-suffix")
+                .arg(&builder.version)
+                .arg("--doc-folder")
+                .arg(builder.doc_out(self.target))
+                .arg("--test-folder")
+                .arg(builder.src.join("src/test/rustdoc-js-parser"));
+            builder.ensure(crate::doc::Std { target: self.target, stage: builder.top_stage });
+            builder.run(&mut command);
+        } else {
+            builder.info("No nodejs found, skipping \"src/test/rustdoc-js-parser\" tests");
```

Comment:
Please do not force-skip this. Make the default condition whether `node` is installed instead.

Reply:
:+1: